core/string.c

```diff
@@ -117,28 +117,36 @@ d_r_strrchr(const char *str, int c)
 /* Private strncpy.  Standard caveat about not copying trailing null byte on
  * truncation applies.
  */
+#define D_R_STRNCPY_BODY()                        \
+    do {                                          \
+        size_t i;                                 \
+        for (i = 0; i < n && src[i] != '\0'; i++) \
+            dst[i] = src[i];                      \
+        /* Pad the rest with nulls. */            \
+        for (; i < n; i++)                        \
+            dst[i] = '\0';                        \
+        return dst;                               \
+    } while (0);
```

Comment:
style: No semicolon here.  Please see all the other macros using do..while (hundreds of them).

Reply:
I actually did and the first two ones I found have a semicolon :)

core/vmareas.c:LOCK_VECTOR
core/vmareas.c:UNLOCK_VECTOR

Do you want me to fix here since we're at it already?

Reply:
None of them have no semicolon in their instantiations.

Reply:
You said

> style: No semicolon here

which I think referred to `} while (0);`

LOCK_VECTOR and UNLOCK_VECTOR have a semicolon there. Most others don't.